File: Code/PatternBattleGame/Assets/Scripts/ObjectManipulation/ObjectRotate.js
var RotationObject : GameObject;

//starting rotation
var RotationX = 0.000;
var RotationY = 0.000;
var RotationZ = 0.000;

var SpeedX = 0.000;
var SpeedY = 0.000;
var SpeedZ = 0.000;

@HideInInspector
var last_turn:boolean = false;
//@HideInInspector
var rotation_disable:boolean = true;

var plateHandler : GameObject;

function Awake(){
	plateHandler = GameObject.Find("Playfield");
}

function start_rotating() {
	last_turn = false;
	rotation_disable = false;	
}

function FixedUpdate () {
	if (rotation_disable)
		return;

	var turn_completed = false;
	if (RotationX > 360) {
		RotationX = 0;
		turn_completed = true;
	}
	if (RotationX < 0) {
		RotationX = 360;
		turn_completed = true;
	}
		

	if (RotationY > 360) {
		RotationY = 0;
		turn_completed = true;
	}
	if (RotationY < 0) {
		turn_completed = true;
		RotationY = 360;
	}

	if (RotationZ > 360) {
		RotationZ = 0;
		turn_completed = true;
	}
	if (RotationZ < 0) {
		RotationZ = 360;
		turn_completed = true;
	}

	if (last_turn && turn_completed) {
		rotation_disable = true;
	//	Debug.Log("done rotating");
		if(plateHandler!=null&&this.GetComponent(GamePlayGUIControl)!=null)
			plateHandler.SendMessage("doneRotating");
	} else {

		RotationX = RotationX+SpeedX;
		//RotationObject.transform.localEulerAngles.x = RotationX;
		RotationY = RotationY+SpeedY;
		//RotationObject.transform.localEulerAngles.y = RotationY;
		RotationZ = RotationZ+SpeedZ;
		//RotationObject.transform.localEulerAngles.z = RotationZ;
	}
	RotationObject.transform.localEulerAngles = Vector3 (RotationX,RotationY,RotationZ);
}

File: Code/PatternBattleGame/Assets/Scripts/ObjectManipulation/ObjectRotate.cs
using JetBrains.Annotations;

using UnityEngine;

public class ObjectRotate : MonoBehaviour
{
    public GameObject rotationObject;

    public float rotationX = 0f;
    public float rotationY = 0f;
    public float rotationZ = 0f;

    public float speedX = 0f;
    public float speedY = 0f;
    public float speedZ = 0f;

    [HideInInspector]
    public bool lastTurn;

    public bool rotationDisable = true;

    public GameObject plateHandler;

    [UsedImplicitly]
    private void Awake()
    {
        plateHandler = GameObject.Find("Playfield");
    }

    [UsedImplicitly]
    private void Update()
    {
        if(rotationDisable)
            return;

        bool turnCompleted = false;
        if (rotationX > 360)
        {
            rotationX = 0;
            turnCompleted = true;
        }
        if (rotationX < 0)
        {
            rotationX = 360;
            turnCompleted = true;
        }

        if (rotationY > 360)
        {
            rotationY = 0;
            turnCompleted = true;
        }
        if (rotationY < 0)
        {
            turnCompleted = true;
            rotationY = 360;
        }

        if (rotationZ > 360)
        {
            rotationZ = 0;
            turnCompleted = true;
        }
        if (rotationZ < 0)
        {
            rotationZ = 360;
            turnCompleted = true;
        }

        if (lastTurn && turnCompleted)
        {
            rotationDisable = true;

            
            if (plateHandler != null && GetComponent<GamePlayGUIControl>() != null)
            {
                plateHandler.SendMessage("doneRotating");
            }
            else
            {
                rotationX = rotationX + speedX;
                rotationY = rotationY + speedY;
                rotationZ = rotationZ + speedZ;
            }
            rotationObject.transform.localEulerAngles = new Vector3(rotationX, rotationY, rotationZ);
        }
    }

    public void start_rotating()
    {
        lastTurn = false;
        rotationDisable = false;
    }


}
